bridged-nft-token/src/nft_core.rs

```diff
@@ -0,0 +1,289 @@
+use crate::*;
+use near_sdk::json_types::{ValidAccountId, U64};
+use near_sdk::{ext_contract, log, Gas, PromiseResult};
+
+const GAS_FOR_NFT_APPROVE: Gas = 10_000_000_000_000;
+const GAS_FOR_RESOLVE_TRANSFER: Gas = 10_000_000_000_000;
+const GAS_FOR_NFT_TRANSFER_CALL: Gas = 25_000_000_000_000 + GAS_FOR_RESOLVE_TRANSFER;
+const NO_DEPOSIT: Balance = 0;
+
+pub trait NonFungibleTokenCore {
+    fn nft_transfer(
+        &mut self,
+        receiver_id: ValidAccountId,
+        token_id: TokenId,
+        approval_id: Option<U64>,
+        memo: Option<String>,
+    );
+
+    /// Returns `true` if the token was transferred from the sender's account.
+    fn nft_transfer_call(
+        &mut self,
+        receiver_id: ValidAccountId,
+        token_id: TokenId,
+        approval_id: Option<U64>,
+        memo: Option<String>,
+        msg: String,
+    ) -> Promise;
+
+    fn nft_approve(&mut self, token_id: TokenId, account_id: ValidAccountId, msg: Option<String>);
+
+    fn nft_revoke(&mut self, token_id: TokenId, account_id: ValidAccountId);
+
+    fn nft_revoke_all(&mut self, token_id: TokenId);
+
+    fn nft_total_supply(&self) -> U64;
```

Comment:
This one is kind of a bummer, but our standards say we'd like to return U128 instead of U64. I think this may apply elsewhere, too. I say it's a bummer because some data structures may not yet have that capability, and we're looking to expand them. So if the data structure has a limit of using U64, we can still cast it into U128 and expect our collections to grow in time. This also applies for `nft_supply_for_owner`, which needs to be implemented as mentioned in previous comments.

Reply:
it was removed and replaced by the one contained inside the standard lib